### PR TITLE
#572 - read groups in AuthFromYaml

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #572 
Updated `http` and corrected `AuthFromYaml` read groups from `credentials.yaml`

### CODE REVIEW

File: src/main/java/com/artipie/auth/AuthFromYaml.java [hunk 1]
```diff
@@ -65,22 +68,18 @@ public AuthFromYaml(final YamlMapping cred) {
...
-                        res = Optional.of(user);
-                    }
-                } else {
-                    if (check(stored, pass)) {
-                        res = Optional.of(user);
-                    }
+                if (type != null && check(stored, type, pass)) {
+                    res = Optional.of(new User(user, AuthFromYaml.groups(users.yamlMapping(user))));
+                } else if (check(stored, pass)) {
+                    res = Optional.of(new User(user, AuthFromYaml.groups(users.yamlMapping(user))));
```
Comment (olegmoz) [inline src/main/java/com/artipie/auth/AuthFromYaml.java:82]: @olenagerasimova merging of nested `if/else` statements you've mad are not identical. Is it intentional? I'd suggest to keep it as it is for now and address in separate issue/PR if it needs to be fixed

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one comment

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected and added todo

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#612` is not assigned, can't get performer

**Pull request merged**